components/Form/CardForm.js

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import styled from "styled-components";
+
+export default function CardForm({
+  onAddCard,
+  onCancel,
+  onPopupContentClick,
+  topics,
+  color,
+}) {
+  const [selectedTopic, setSelectedTopic] = useState("");
+  const [question, setQuestion] = useState("");
+  const [answer, setAnswer] = useState("");
+  const [questionError, setQuestionError] = useState(false);
+  const [answerError, setAnswerError] = useState(false);
+
+  const handleTopicChange = (event) => {
+    const value = event.target.value;
+    setSelectedTopic(value);
+  };
+
+  const handleQuestionChange = (event) => {
+    const value = event.target.value;
+    setQuestion(value);
+  };
+
+  const handleAnswerChange = (event) => {
+    const value = event.target.value;
+    setAnswer(value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (question.trim() === "" || answer.trim() === "") {
+      setQuestionError(question.trim() === "");
+      setAnswerError(answer.trim() === "");
```

Comment:
You already proof the condition inside the if condition so you don't need to check the condition agian inside the setter functions.

```suggestion
    if (question.trim() === "" || answer.trim() === "") {
      setQuestionError(true);
      setAnswerError(true);
```

Reply:
Then you could delete the if condition. The point is to avoid dopple checks.

Reply:
When I remove the if statement I can not escape the function with the return statements inside of if. That is why encapsulated them inside an if statement, to return if either of them are true and then set the right errors to true. Do you have another suggestion how to solve this?

Reply:
Okay, dann you should split the if statement in two if conditions.